src/go/configgenerator/listener_generator.go

```diff
@@ -165,6 +166,33 @@ func MakeListener(serviceInfo *sc.ServiceInfo, filterGenerators []*filterconfig.
 	return listener, nil
 }
 
+// To fix b/221072669: a hack to work around b/221308324 where
+// Cloud Run always set :scheme header to http when using http2 protocol for grpc.
+// Override scheme header to https when following conditions meet:
+// * Deployed in serverless platform.
+// * Backend uses grpc
+// * All remote backends are using TLS
+func makeSchemeHeaderOverride(serviceInfo *sc.ServiceInfo) *corepb.SchemeHeaderTransformation {
+	if serviceInfo.Options.ComputePlatformOverride != util.ServerlessPlatform || !serviceInfo.GrpcSupportRequired {
+		return nil
+	}
+	allTLS := true
+	for _, v := range serviceInfo.RemoteBackendClusters {
```

Comment:
What will happen if some backends are TLS but some aren't?

Reply:
We could not use scheme_override.  For now, envoy scheme_override applies to all backends.  if some backend is not ssl,  the override will be wrong. 

I think this is ok.  All backends for serverless are using TLS.

Reply:
> All backends for serverless are using TLS.

If this is the case, I think we should just return error here so that it simply fails at startup?

Reply:
That will be too brutal. In the case of not all backends use TLS,  my pr just don't do "override_scheme=https" ,  your proposal is, not to support it at all.  

I am afraid that there are some users using ESPv2 in such way.  We don't want to break them.  They may not need correct "scheme" header.   Only IAP backends require correct "scheme" headers.

Reply:
I see. Where I come from is that it looks to me there are some combinations that we just cannot support. If there are some backends require https like IAP and some backends require http? Anyway we'll have to break one of them?

Reply:
Just discussed with Wayne about this. Since Envoy doesn't allow per backend scheme override, it would be a breaking change for certain use cases. Now, we only override scheme when makeSchemeHeaderOverride **all** backends are https which breaks the case Shuo mentioned where some backends are using IAP and some backends are HTTP.
. For fixing that, probably we can change to override scheme when **any** backend is https. Most Cloud Run backend should be https and the http backend should be fine with this wrong scheme header.

Reply:
>Most Cloud Run backend should be https and the http backend should be fine with this wrong scheme header
Thanks. This sounds good.

Reply:
Code is updated to:  if ANY of backends is using SSL,  we will use scheme_override.  This is to supported multiple backends with mixed SSL and non-SSL.